components/molecule/buttonGroup/src/index.js

```diff
@@ -25,10 +25,17 @@ const MoleculeButtonGroup = ({
     groupPositions,
     numChildren
   )
-  const extendedChildren = React.Children.map(children, (child, index) => {
-    const groupPosition = getGroupPositionByIndex(index)
-    return React.cloneElement(child, {...props, type, groupPosition, fullWidth})
-  })
+  const extendedChildren = React.Children.toArray(children)
+    .filter(child => !!child)
```

Comment:
You could write this like `.filter(Boolean)` is the same but look like a little cleaner